azkaban-common/src/main/java/azkaban/AzkabanCommonModule.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package azkaban;
+
+import azkaban.project.JdbcProjectLoader;
+import azkaban.project.ProjectLoader;
+import azkaban.spi.Storage;
+import azkaban.spi.StorageException;
+import azkaban.storage.LocalStorage;
+import azkaban.storage.StorageConfig;
+import azkaban.storage.StorageImplementationType;
+import azkaban.utils.Props;
+import com.google.inject.AbstractModule;
+import com.google.inject.Inject;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import java.io.File;
+
+import static azkaban.storage.StorageImplementationType.*;
+
+
+public class AzkabanCommonModule extends AbstractModule {
+  private final Props props;
+  /**
+   * Storage Implementation
+   * This can be any of the {@link StorageImplementationType} values in which case {@link StorageFactory} will create
+   * the appropriate storage instance. Or one can feed in a custom implementation class using the full qualified
+   * path required by a classloader.
+   *
+   * examples: LOCAL, DATABASE, azkaban.storage.MyFavStorage
```

Comment:
I went with that approach initially. I agree that would simplify the code and can get rid of the enum, etc. The downside is that a hardcoded class reference is difficult to refactor. 

Tomorrow if we decide to rename or move the class all configs need to change. This might be important today since the codebase is going through a lot of churns.
Secondly, the name mapping kind of draws a strict line between what is supported out of the box versus what is supported externally.

What do you think ?